internal/topic/validation_test.go

```diff
@@ -46,4 +46,33 @@ func TestValidation_GetValidatedModel(t *testing.T) {
 	})
 	assert.Nil(t, err)
 	assert.Equal(t, m.Name, topic)
+	assert.Equal(t, m.NumPartitions, DefaultNumPartitions)
+}
+
+func TestValidation_GetValidatedAdminModel(t *testing.T) {
+	ctx := context.Background()
+
+	topic := "projects/test-project/topics/test-topic"
+	numPartitions := 5
+	m, err := GetValidatedTopicForAdminUpdate(ctx, &metrov1.AdminTopic{
```

Comment:
for starting we should just allow update of `NumPartitions` ?

Reply:
Currently, only `NumPartitions` gets updated. `Name` is only used to identify the topic. Kept `Labels` for compatibility with the existing `Topic` model.